internal/controller/providers/pg/provider.go

```diff
@@ -109,6 +111,29 @@ func (p *Provider) Status(ctx context.Context) (everestv1alpha1.DatabaseClusterS
 		status.Status = everestv1alpha1.AppStateRestoring
 	}
 
+	// Get PG pods to check if an upgrade is pending or in progress.
+	listOpts := &client.ListOptions{
+		Namespace:     p.DB.GetNamespace(),
+		LabelSelector: labels.SelectorFromSet(labels.Set{"app.kubernetes.io/component": "pg", "app.kubernetes.io/instance": p.DB.GetName()}),
+	}
+	podList := &corev1.PodList{}
+	if err := c.List(ctx, podList, listOpts); err != nil {
+		return status, err
+	}
+	for _, pod := range podList.Items {
+		for _, container := range pod.Spec.Containers {
+			if container.Name != "database" {
+				continue
+			}
+
+			// If there's a pod with a different image tag from the one
+			// specified in the CR, an upgrade is pending or in progress.
+			if container.Image != pg.Spec.Image {
+				status.Status = everestv1alpha1.AppStateUpgrading
```

Comment:
it seems it will be enough to get the first tags mismatch in order to make a decision that DB cluster is in `upgrading` state. No need to check the rest of pods (especially in big clusters)

Reply:
and I would move this logic to a separate function like `isDatabseClusterInUpgradeState` or similar. the same as we have above `common.IsDatabaseClusterRestoreRunning`

Reply:
Agree, done